src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        if (this[i] === null || this[i] === undefined) {
+          break;
+        }
+        str += this[i].toString();
+        break;
+      }
```

Comment:
I think this code needs improvement. To match if and big nesting. In my mind you should think  how do you do without this nesting. It is 'KOSTULY' 😁😂. Please try to separate the conditions for the separator and the element :) I think it will be better)